packages/client/src/components/Modals/SearchPanel.vue

```diff
@@ -159,26 +158,43 @@ export default {
   },
   computed: {
     ...mapGetters({
-      fetchEligibilityCodes: 'grants/fetchEligibilityCodes',
+      searchFormFilters: 'grants/searchFormFilters',
+      eligibilityCodes: 'grants/eligibilityCodes',
     }),
   },
   methods: {
     ...mapActions({
       createSavedSearch: 'grants/createSavedSearch',
       updateSavedSearch: 'grants/updateSavedSearch',
+      applyFilters: 'grants/applyFilters',
+      fetchEligibilityCodes: 'grants/fetchEligibilityCodes',
     }),
     setup() {
-      // this.fetchEligibilityCodes();
+      this.fetchEligibilityCodes();
     },
     customLimitText(count) {
       return `+${count}`;
     },
     eligibilityLabel({ label }) {
       return label;
     },
+    apply() {
+      const formDataCopy = { ...this.formData.criteria };
+      this.applyFilters(formDataCopy);
+      this.$emit('filters-applied');
+      this.$refs.searchPanelSideBar.hide();
+    },
+    syncFilterState() {
+      this.formData.criteria = { ...this.searchFormFilters };
+    },
+    onShown() {
+      // current filters may have changed since form was opened
+      this.syncFilterState();
+    },
     saveSearch() {
       console.log('foo');
       console.log(this.formData);
+      this.apply();
```

Comment:
I think eventually we'd call this after the updateSavedSearch call finishes/promise resolves, but that's currently throwing an error for me, so I just dropped this inline here for now. Does that sound right?

Reply:
Yup, this looks right to me. Currently we don't have an edit endpoint just yet, but once it is there it should work.